File: solutions/fable-1/fable/python/roman.py
import sys

PAIRS = [
    (1000, "M"), (900, "CM"), (500, "D"), (400, "CD"),
    (100, "C"), (90, "XC"), (50, "L"), (40, "XL"),
    (10, "X"), (9, "IX"), (5, "V"), (4, "IV"), (1, "I"),
]


def to_roman(n):
    out = []
    for value, sym in PAIRS:
        while n >= value:
            out.append(sym)
            n -= value
    return "".join(out)


def main():
    lines = []
    for line in sys.stdin:
        s = line.strip()
        if s:
            lines.append(to_roman(int(s)))
    sys.stdout.write("\n".join(lines) + ("\n" if lines else ""))


if __name__ == "__main__":
    main()

File: solutions/fable-1/fable/typescript/roman.ts
const input = await Bun.stdin.text();

const table: [number, string][] = [
  [1000, "M"],
  [900, "CM"],
  [500, "D"],
  [400, "CD"],
  [100, "C"],
  [90, "XC"],
  [50, "L"],
  [40, "XL"],
  [10, "X"],
  [9, "IX"],
  [5, "V"],
  [4, "IV"],
  [1, "I"],
];

function toRoman(n: number): string {
  let out = "";
  for (const [value, sym] of table) {
    while (n >= value) {
      out += sym;
      n -= value;
    }
  }
  return out;
}

const lines = input.split(/\r?\n/);
const results: string[] = [];
for (const line of lines) {
  const s = line.trim();
  if (s === "") continue;
  const n = parseInt(s, 10);
  if (Number.isNaN(n)) continue;
  results.push(toRoman(n));
}

process.stdout.write(results.join("\n") + (results.length ? "\n" : ""));
